### PR TITLE
[WIP] Enable HeatmapLayer Support

### PR DESCRIPTION
**Pull request by dhee9000**
Related to issue: @react-native-mapbox-gl/maps#115

Adds support for HeatmapLayer in Style Layers for Android and iOS.

- On Android, Heatmap Layer appears to be working, at least for the examples provided for the native Mapbox SDK
  - RCTMGLHeatmapLayer.java created based on RCTMGLCircleLayer
  - RCTMGLHeatmapLayerManager.java based on RCTMGLCircleLayerManager
  - Styles were already present under RCTMGLStyle.java
  - Added to RCTMGLHeatmapLayerManager to RCTMGLPackage so that it works in React Native

- On iOS, the project does not run, possibly missing changes higher up in the library.
  - RCTMLGHeatmapLayer.m and .h created based on RCTMGLCircleLayer.m and h
  - RCTMGLHeatmapLayerManager.m and .h created based on RCTMGLCircleLayerManager.m and h

TODO:
Test style properties and expressions for HeatmapLayer
Add example to example project
Complete documentation of available properties

### CODE REVIEW

**Comment by dhee9000** [comment]
After these commits, iOS works but the use of the heatmapDensity expression for the heatmapColor property causes the app to crash.

File: docs/HeatmapLayer.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+## <MapboxGL.HeatmapLayer />
```
Comment (mfazekas) [inline docs/HeatmapLayer.md]: Was this doc generated? Normally docs are generated by `npm run generate`

**Comment by mfazekas** [comment]
> After these commits, iOS works but the use of the heatmapDensity expression for the heatmapColor property causes the app to crash.

@dhee9000 thanks awesome work! Can you add a stacktrace? Or even a heatmap sample to the example app?

File: docs/HeatmapLayer.md (hunk 1 above)
Comment (dhee9000) [reply docs/HeatmapLayer.md]: No, I didn't know this script existed, I manually copied over another doc and put together data from the site. I will generate it with this and add another commit.

**Comment by mfazekas** [comment]
@dhee9000 thanks for the PR, i've added a followup pr with a simple heatmap example - #136

**Pull request merged**